apps/web/server/routers/viewer.tsx

```diff
@@ -1040,14 +1041,41 @@ const loggedInViewerRouter = createProtectedRouter()
           }
         }
 
-        // If it's a calendar, remove the destination claendar from the event type
+        // If it's a calendar, remove the destination calendar from the event type
         if (credential.app?.categories.includes(AppCategories.calendar)) {
           if (eventType.destinationCalendar?.integration === credential.type) {
-            await prisma.destinationCalendar.delete({
+            const destinationCalendar = await prisma.destinationCalendar.findFirst({
               where: {
-                id: eventType.destinationCalendar.id,
+                id: eventType.destinationCalendar?.id,
               },
             });
+            if (destinationCalendar) {
+              await prisma.destinationCalendar.delete({
+                where: {
+                  id: destinationCalendar.id,
+                },
+              });
+            }
+          }
+
+          if (externalId) {
+            const existingSelectedCalendar = await prisma.selectedCalendar.findFirst({
+              where: {
+                externalId: externalId,
+              },
+            });
+            // @TODO: SelectedCalendar doesn't have unique ID so we should only delete one item
+            if (existingSelectedCalendar) {
```

Comment:
This was necessary because after disconnecting a calendar it will remain on selectedCalendars.